### PR TITLE
AggregatedCompliant_EVA-466

### PR DESCRIPTION
**Pull request by poggio84**

### CODE REVIEW

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java [hunk 1]
```diff
@@ -21,59 +21,81 @@
...
+    public static final String NORMALIZE_VARIANTS = "Normalize variants";
+    public static final String LOAD_VARIANTS = "Load variants";
+
+    //Job default values
+    {
+        includeSamples = false;
+        compressGenotypes = false;
+        calculateStats = true;
+        includeStats = true;
+    }
```
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java]: Wasn't it agreed to encapsulate these fields into small POJOs that would be injected from a job-specific `@Bean` method? Did that cause any issues?

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java [hunk 2]
```diff
@@ -21,59 +21,81 @@
...
 import org.springframework.context.annotation.Import;
 
+import org.springframework.context.annotation.Scope;
 import uk.ac.ebi.eva.pipeline.jobs.steps.VariantsLoad;
 import uk.ac.ebi.eva.pipeline.jobs.steps.VariantsTransform;
 
 /**
- * Job similar to VariantConfiguration, but it takes an aggregated VCF,
- * which has no sample information (e.g. no genotypes).
+ *  Complete pipeline workflow for aggregated Vcf.
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java]: Opss.. fixed in next commit

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java [hunk 3]
```diff
@@ -21,59 +21,81 @@
...
+        return builder.build();
     }
 
-    private Step transform() {
-        return generateStep(NORMALIZE_VARIANTS,variantsTransform);
+    @Bean
+    @Scope("prototype")
+    protected Step transform() {
+        return generateStep(NORMALIZE_VARIANTS,
+                new VariantsTransform(updateVariantOptionsWithJobSpecificValues(getVariantOptions()), getPipelineOptions()));
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java]: Moved in @PostConstruct

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/CommonJobStepInitialization.java [hunk 4]
```diff
@@ -60,7 +66,22 @@ protected Step generateStep(String stepName, Tasklet tasklet) {
         return taskletBuilder.build();
     }
 
+    protected ObjectMap updateVariantOptionsWithJobSpecificValues (ObjectMap variantOptions){
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/CommonJobStepInitialization.java]: Created two new methods in `VariantJobsArgs` as suggested

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java (hunk 1 above)
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfiguration.java]: Fields are now initialized in `VariantJobsArgs` like all the others

File: src/main/java/uk/ac/ebi/eva/pipeline/jobs/CommonJobStepInitialization.java (hunk 4 above)
Comment (cyenyxe) [inline src/main/java/uk/ac/ebi/eva/pipeline/jobs/CommonJobStepInitialization.java]: Sorry I explained myself pretty badly: my idea was to configure how samples are stored (INCLUDE_SAMPLES and COMPRESS_GENOTYPES) via `configureGenotypesStorage`, whereas the statistics storage (CALCULATE_STATS and INCLUDE_STATS) would be via `configureStatisticsStorage`. Those methods would then receive a couple of boolean arguments each, without being aware of which values each job class requires.

File: src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoadTest.java [hunk 5]
```diff
@@ -0,0 +1,166 @@
...
+import static uk.ac.ebi.eva.test.utils.JobTestUtils.getLines;
+
+/**
+ * @author Diego Poggioli
+ *
+ * Test for {@link VariantsLoad}
+ */
+@RunWith(SpringJUnit4ClassRunner.class)
+@ContextConfiguration(classes = {VariantConfiguration.class, VariantJobsArgs.class, VariantConfig.class})
+public class VariantLoadTest {
```
Comment (poggio84) [inline src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoadTest.java]: Tasklets at the moment are all tested as step (.launchStep()). I would probably have a second look at `VariantStatsConfigurationTest` that is containing .launchStep() even if it is supposed to be a test for Job. I will check if I can externalize this..

File: src/test/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfigurationTest.java [hunk 6]
```diff
@@ -18,188 +18,153 @@
...
+    public void aggregatedTransformAndLoadShouldBeExecuted() throws Exception {
+        Config.setOpenCGAHome(opencgaHome);
 
-        pipelineOptions.put("input.vcf", input);
-        variantOptions.put(VariantStorageManager.DB_NAME, dbName);
+        // transformedVcf file init
+        String transformedVcf = outputDir + input + ".variants.json" + compressExtension;
+        File transformedVcfFile = new File(transformedVcf);
+        transformedVcfFile.delete();
+        assertFalse(transformedVcfFile.exists());
```
Comment (poggio84) [inline src/test/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfigurationTest.java]: Removed

File: src/test/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfigurationTest.java [hunk 7]
```diff
@@ -18,188 +18,153 @@
...
-
-        pipelineOptions.put("input.vcf", input);
-        variantOptions.put(VariantStorageManager.DB_NAME, dbName);
-
-        variantOptions.put("includeStats", true);
-
-        VariantSource source = (VariantSource) variantOptions.get(VariantStorageManager.VARIANT_SOURCE);
-
-        variantOptions.put(VariantStorageManager.VARIANT_SOURCE, new VariantSource(
+    public void aggregationNoneOptionShouldNotLoadStats() throws Exception {
```
Comment (poggio84) [inline src/test/java/uk/ac/ebi/eva/pipeline/jobs/VariantAggregatedConfigurationTest.java]: In this case is checking that the stats are not loaded (.getCohortStats().isEmpty()) when we use Aggregation.NONE over an aggregated VCF. If this is not important or doesn't make sense then I suppose I can delete the test

File: src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoadTest.java (hunk 5 above)
Comment (poggio84) [inline src/test/java/uk/ac/ebi/eva/pipeline/jobs/steps/VariantLoadTest.java]: Step tests in `VariantStatsConfigurationTest` extracted in `PopulationStatisticsGeneratorStepTest` and `PopulationStatisticsLoaderStepTest`

File: src/main/java/uk/ac/ebi/eva/pipeline/configuration/VariantJobsArgs.java [hunk 8]
```diff
@@ -240,6 +231,16 @@ private void loadPipelineOptions() {
         logger.debug("Using as pipelineOptions: {}", pipelineOptions.entrySet().toString());
     }
 
+    public void configureGenotypesStorage (boolean include_samples, boolean compress_genotypes){
```
Comment (poggio84) [inline src/main/java/uk/ac/ebi/eva/pipeline/configuration/VariantJobsArgs.java]: Ahh bad copy and paste (facepalm) fixed :)

**Pull request merged**